### PR TITLE
Linux: Add support for MouseButton::Navigate in GPUI (wayland and x11)

### PR DESCRIPTION
**Pull request by bbb651**
Release Notes:

- N/A

Based on wgpu implementation (which I wrote).

### CODE REVIEW

**Comment by bbb651** [comment]
@cla-bot check

File: crates/gpui/src/platform/linux/wayland/client.rs [hunk 1]
```diff
@@ -23,12 +23,12 @@ use crate::platform::linux::client::Client;
 use crate::platform::linux::wayland::window::WaylandWindow;
 use crate::platform::{LinuxPlatformInner, PlatformWindow};
 use crate::PlatformInput::KeyDown;
-use crate::ScrollDelta;
 use crate::{
     platform::linux::wayland::window::WaylandWindowState, AnyWindowHandle, DisplayId, KeyDownEvent,
     KeyUpEvent, Keystroke, Modifiers, MouseButton, MouseDownEvent, MouseMoveEvent, MouseUpEvent,
     Pixels, PlatformDisplay, PlatformInput, Point, ScrollWheelEvent, TouchPhase, WindowOptions,
 };
+use crate::{NavigationDirection, ScrollDelta};
```
Comment (kvark) [inline crates/gpui/src/platform/linux/wayland/client.rs:31]: nit: should be in the same `use` section with above

File: crates/gpui/src/platform/linux/wayland/client.rs [hunk 2]
```diff
@@ -452,13 +452,24 @@ impl Dispatch<wl_keyboard::WlKeyboard, ()> for WaylandClientState {
...
+    const BTN_EXTRA: u32 = 0x114;
+    const BTN_FORWARD: u32 = 0x115;
+    const BTN_BACK: u32 = 0x116;
+
+    Some(match button {
+        BTN_LEFT => MouseButton::Left,
+        BTN_RIGHT => MouseButton::Right,
+        BTN_MIDDLE => MouseButton::Middle,
+        BTN_BACK | BTN_SIDE => MouseButton::Navigate(NavigationDirection::Back),
+        BTN_FORWARD | BTN_EXTRA => MouseButton::Navigate(NavigationDirection::Back),
```
Comment (kvark) [inline crates/gpui/src/platform/linux/wayland/client.rs:470]: this should be `NavigationDirection::Forward`

File: crates/gpui/src/platform/linux/wayland/client.rs [hunk 3]
```diff
@@ -516,11 +527,14 @@ impl Dispatch<wl_pointer::WlPointer, ()> for WaylandClientState {
                 if let (Some(focused_window), Some(mouse_location)) =
                     (focused_window, mouse_location)
                 {
+                    let Some(button) = linux_button_to_gpui(button) else {
```
Comment (kvark) [inline crates/gpui/src/platform/linux/wayland/client.rs:530]: First I'm seeing this pattern. Out of curiosity, is it idiomatic?

**Review by kvark — COMMENTED** [review]
Nice work!

**Pull request merged**